opencti-platform/opencti-front/src/private/components/dataEntities/data/DataSourceCreation.js

```diff
@@ -0,0 +1,370 @@
+    position: 'fixed',
+    bottom: 30,
+    right: 30,
+  },
+  drawerPaper: {
+    minHeight: '100vh',
+    width: '50%',
+    position: 'fixed',
+    overflow: 'auto',
+    backgroundColor: theme.palette.navAlt.background,
+    transition: theme.transitions.create('width', {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+    padding: 0,
+  },
+});
+
+const dataSourceCreationMutation = graphql`
+  mutation DataSourceCreationMutation($input: RiskAddInput) {
+    createRisk (input: $input) {
+      id
+      # ...RiskCard_node
+      # ...RiskDetails_risk
+      # operational_status
+      # serial_number
+      # release_date
+      # description
+      # version
+      # name
+    }
+  }
+`;
+
+const riskValidation = (t) => Yup.object().shape({
+  name: Yup.string().required(t('This field is required')),
+  // asset_type: Yup.array().required(t('This field is required')),
+  // implementation_point: Yup.string().required(t('This field is required')),
+  // operational_status: Yup.string().required(t('This field is required')),
+  // first_seen: Yup.date()
+  //   .nullable()
+  //   .typeError(t('The value must be a date (YYYY-MM-DD)')),
+  // last_seen: Yup.date()
+  //   .nullable()
+  //   .typeError(t('The value must be a date (YYYY-MM-DD)')),
+  // sophistication: Yup.string().nullable(),
+  // resource_level: Yup.string().nullable(),
+  // primary_motivation: Yup.string().nullable(),
+  // secondary_motivations: Yup.array().nullable(),
+  // personal_motivations: Yup.array().nullable(),
+  // goals: Yup.string().nullable(),
+});
+const Transition = React.forwardRef((props, ref) => (
+  <Slide direction="up" ref={ref} {...props} />
+));
+Transition.displayName = 'TransitionSlide';
+class DataSourceCreation extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      onSubmit: false,
+      displayCancel: false,
+    };
+  }
+
+  handleCancelButton() {
+    this.setState({ displayCancel: false });
+  }
+
+  handleOpenCancelButton() {
+    this.setState({ displayCancel: true });
+  }
+
+  handleOpen() {
+    this.setState({ open: true });
+  }
+
+  onSubmit(values, { setSubmitting, resetForm }) {
+    const adaptedValues = evolve(
+      {
+        deadline: () => parse(values.deadline).format(),
+      },
+      values,
+    );
+    const finalValues = R.pipe(
+      R.dissoc('controls'),
+      R.assoc('name', values.name),
+      R.dissoc('riskDetailsCreated'),
+      R.dissoc('riskDetailsModified'),
+    )(adaptedValues);
+    CM(environmentDarkLight, {
+      mutation: dataSourceCreationMutation,
+      variables: {
+        input: finalValues,
+      },
+      setSubmitting,
+      onCompleted: (data) => {
+        setSubmitting(false);
+        resetForm();
+        this.handleClose();
+        // this.props.history.push('/activities/risk assessment/risks');
+      },
+      onError: (err) => console.error(err),
```

Comment:
@sachin755 please apply an error toast here.

Reply:
#299 is just for the design of main screen.  The chages you've suggested are going to be inside #302